### PR TITLE
Add names of attributes translations

### LINKED ISSUE #2400: Product attributes in translations should be named
### User story:

I'm trying to translate product attributes:

1. Go to translations
2. Choose any language
3. Go to products
4. Choose a product with attributes
5. Atributes are named `Attribute 1`, `Attribute 2`

### Acceptance criteria:

- Attributes should be named e.g. `Attribute Material` so that we know which attribute we're translating.
- Plain text attribute translations also should be handled in this view, not only rich text

### PR DESCRIPTION
**Pull request by orzechdev**
It introduces in translation views (product, variant and page translation views) the following:
- replaced numbers with names of attributes,
- added the possibility to translate plain text attributes (not only rich text).

Fixes #2400.

Backend: https://github.com/saleor/saleor/pull/11028

Example to test: https://2400-add-names-of-attributes-translations.dashboard.saleor.rocks/translations/EL/products/UHJvZHVjdDo3Mg%3D%3D

**PR intended to be tested with API branch:** add-attribute-field-to-AttributeValueTranslatableContent-type

### Screenshots

Screenshot 2022-10-24 at 17 59 10

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://add-attribute-field-to-attributevaluetranslatablecontent-type.api.saleor.rocks/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [x] translations
21. [ ] navigation
22. [ ] variants

CONTAINERS=1

### CODE REVIEW

File: src/translations/components/TranslationsProductVariantsPage/TranslationsProductVariantsPage.tsx [hunk 1]
```diff
@@ -117,22 +117,27 @@ const TranslationsProductsPage: React.FC<TranslationsProductsPageProps> = ({
...
-                type: "rich" as "rich",
-                value: attrVal?.richText,
+                name: attrVal.name,
+                translation:
+                  attrVal.translation?.richText ||
+                  attrVal.translation?.plainText ||
+                  null,
+                type: attrVal.richText
+                  ? ("rich" as "rich")
+                  : ("short" as "short"),
```
Comment (andrzejewsky) [inline src/translations/components/TranslationsProductVariantsPage/TranslationsProductVariantsPage.tsx:139]: Could you properly type this?

**Review by andrzejewsky — CHANGES_REQUESTED** [review]
Please try to avoid "as" assertions, if you cannot, please create a function with that but with returning expected type

File: src/translations/components/TranslationsProductVariantsPage/TranslationsProductVariantsPage.tsx (hunk 1 above)
Comment (orzechdev) [reply src/translations/components/TranslationsProductVariantsPage/TranslationsProductVariantsPage.tsx]: Sure, fixed

**Review by krzysztofzuraw — APPROVED** [review]

File: src/translations/components/TranslationsProductsPage/TranslationsProductsPage.tsx [hunk 2]
```diff
@@ -163,22 +164,25 @@ const TranslationsProductsPage: React.FC<TranslationsProductsPageProps> = ({
...
                 ),
-                name: attrVal?.name,
-                translation: attrVal?.translation?.richText || null,
-                type: "rich" as "rich",
-                value: attrVal?.richText,
+                name: attrVal.name,
+                translation:
+                  attrVal.translation?.richText ||
+                  attrVal.translation?.plainText ||
+                  null,
```
Comment (andrzejewsky) [inline src/translations/components/TranslationsProductsPage/TranslationsProductsPage.tsx:183]: so, you repeat the same logic, a function perhaps, and reuse it?

**Review by andrzejewsky — DISMISSED** [review]

File: src/translations/components/TranslationsProductVariantsPage/TranslationsProductVariantsPage.tsx [hunk 3]
```diff
@@ -117,22 +118,25 @@ const TranslationsProductsPage: React.FC<TranslationsProductsPageProps> = ({
...
                 ),
-                name: attrVal?.name,
-                translation: attrVal?.translation?.richText || null,
-                type: "rich" as "rich",
-                value: attrVal?.richText,
+                name: attrVal.name,
+                translation:
+                  attrVal.translation?.richText ||
+                  attrVal.translation?.plainText ||
+                  null,
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/translations/components/TranslationsProductVariantsPage/TranslationsProductVariantsPage.tsx]: Yes, it is, I checked it and it's not needed.

File: src/translations/components/TranslationsProductsPage/TranslationsProductsPage.tsx (hunk 2 above)
Comment (orzechdev) [reply src/translations/components/TranslationsProductsPage/TranslationsProductsPage.tsx]: Good idea, I added a function for it.

File: src/translations/types.ts [hunk 4]
```diff
@@ -19,12 +19,14 @@ export enum PageTranslationInputFieldName {
   richText = "richText",
 }
 
+export type TranslationFieldType = "short" | "long" | "rich";
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/translations/types.ts]: Sure, might be better with constant object.

[the remark this answers is not in the document]
Comment (orzechdev) [reply src/translations/types.ts]: Just for further reference, there was discussion few months ago, about enums that we would prefer string literals and constant object with strings as you proposed in your comment. Enums have some drawbacks, may not be really type-safe, introduce excessive code into bundle size or require importing them in every moment we use them https://fettblog.eu/tidy-typescript-avoid-enums/

File: src/translations/utils.ts [hunk 5]
```diff
@@ -52,3 +59,34 @@ export const getTranslationFields = (
...
+      attrVal.translation?.richText || attrVal.translation?.plainText,
+    type: attrVal.richText ? "rich" : "short",
+    value: attrVal.richText || attrVal.plainText,
+  })) || [];
+
+export const getAttributeValueTranslationsInputData = (
+  type: TranslationFieldType,
+  data: OutputData | string,
+): AttributeValueTranslationInput =>
+  type === "rich"
```
Comment (krzysztofzuraw) [inline src/translations/utils.ts:90]:
```suggestion
  type === TranslationFieldType.RICH
```

**Review by andrzejewsky — DISMISSED** [review]

File: src/translations/utils.ts (hunk 5 above)
Comment (orzechdev) [reply src/translations/utils.ts]: 👍 Fixed

**Review by krzysztofzuraw — APPROVED** [review]

**Comment by orzechdev** [comment]
@szczecha fixed.
@andrzejewsky There is a component `TranslationFields` which expects explicitly only `undefined` to show skeleton. Rich text although displays skeleton based on disabled, but other fields not. So I left null explicitly in `translation: attrVal.translation?.richText || attrVal.translation?.plainText || null,` to leave skeleton working as it was before.

**Review by szczecha — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**